storagenode/notifications/notifications.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package notifications
+
+import (
+	"context"
+	"time"
+
+	"github.com/skyrings/skyring-common/tools/uuid"
+
+	"storj.io/storj/pkg/storj"
+)
+
+// DB tells how application works with notifications database.
+//
+// architecture: Database
+type DB interface {
+	Insert(ctx context.Context, notification NewNotification) (Notification, error)
+	List(ctx context.Context, cursor NotificationCursor) (NotificationPage, error)
+	Read(ctx context.Context, notificationID uuid.UUID) error
+	ReadAll(ctx context.Context) error
+}
+
+// NotificationType is a numeric value of specific notification type.
+type NotificationType int
+
+const (
+	// NotificationTypeCustom is a common notification type which doesn't describe node's core functionality.
+	// TODO: change type name when all notification types will be known
+	NotificationTypeCustom NotificationType = 0
+	// NotificationTypeAuditCheckFailure is a notification type which describes node's audit check failure.
+	NotificationTypeAuditCheckFailure NotificationType = 1
+	// NotificationTypeUptimeCheckFailure is a notification type which describes node's uptime check failure.
+	NotificationTypeUptimeCheckFailure NotificationType = 2
+	// NotificationTypeDisqualification is a notification type which describes node's disqualification status.
+	NotificationTypeDisqualification NotificationType = 3
+)
+
+// NewNotification holds notification entity info which is being received from satellite or local client.
+type NewNotification struct {
+	SenderID storj.NodeID
+	Type     NotificationType
```

Comment:
Can notifications have only one type? Maybe it should be tags?

Reply:
what do you mean? could you please provide an example?

Reply:
For example, could the notification be a `warning` and `audit`, or something similar? Currently it seems we end up with tons of constants that all try to encode different ideas into a single enumeration. Usually, the level and scope are separated. For some logging/notification systems there's support for custom tags.

Reply:
hm, yeah. So you think we should add fields for level and tags?

Reply:
It's more of a question, . I.e. I don't have a strong argument either way, but I usually just have seen them separated, mainly because it's more flexible.